src/test/java/org/folio/rest/support/matchers/DomainEventAssertions.java

```diff
@@ -90,42 +101,38 @@ private static void assertHeaders(List<KafkaHeader> headers) {
   }
 
   public static void assertNoUpdateEvent(String instanceId) {
-    await().atLeast(1, SECONDS);
-    await().atMost(1, SECONDS)
-      .until(() -> getInstanceEvents(instanceId).size(), greaterThan(0));
+    await()
+      .until(() -> getInstanceEvents(instanceId), is(not(empty())));
 
     final JsonObject updateMessage  = getLastInstanceEvent(instanceId).value();
     assertThat(updateMessage.getString("type"), not(is("UPDATE")));
   }
 
   public static void assertNoUpdateEventForHolding(String instanceId, String hrId) {
-    await().atLeast(1, SECONDS);
-    await().atMost(1, SECONDS)
-      .until(() -> getHoldingsEvents(instanceId, hrId).size(), greaterThan(0));
+    await()
+      .until(() -> getHoldingsEvents(instanceId, hrId), is(not(empty())));
 
     final JsonObject updateMessage  = getLastHoldingEvent(instanceId, hrId).value();
     assertThat(updateMessage.getString("type"), not(is("UPDATE")));
   }
 
   public static void assertNoRemoveEvent(String instanceId) {
-    await().atLeast(1, SECONDS);
-    await().atMost(1, SECONDS)
-      .until(() -> getInstanceEvents(instanceId).size(), greaterThan(0));
+    await()
+      .until(() -> getInstanceEvents(instanceId), is(not(empty())));
```

Comment:
> await().until(someCondition); will succeed if the condition is or becomes true before the await() timeout, otherwise it fails. A single point of time where condition is true is sufficient. The await() timeout is 2 seconds.

I think this eagerness makes `until` (without a minimum wait) to be inappropriate for checking for the absence of something, as it will falsely succeed immediately.

Reply:
This test doesn't check the absence of a delete event, but the presence of an event, and that that event is not a delete event.

Reply:
Hmm, I don't think the logic of this assertion makes sense (and that might be my mistake as I may have changed it previously).

The first event not being of type `DELETE` does not mean that no message of type `DELETE` has been published, which is what the name `assertNoRemoveEvent`.

If this is beyond the scope of this pull request, that is ok, it isn't caused by your change.